Java/other/KnightsTour.java

```diff
@@ -0,0 +1,413 @@
+
+import java.io.*;
+import java.util.*;
+public class KnightsTour {
+    public static void main(String[] args) throws Exception {
+        Scanner scn = new Scanner(System.in);
+        int n = scn.nextInt();
+        int[][] chess = new int[n][n];
+        int r = scn.nextInt();
+        int c = scn.nextInt();
+             //call the function
+        knight( chess, r, c, 1 );      
+    }
+
+    public static void knight (int[][] chess, int r, int c, int jump) {
+             //base case of recursion when the jump becomes equal to the length of check board
+         if( jump == chess.length * chess[0].length ){
+            //assign the jump number on the chess board
+            chess[r][c] = jump;
+             //call to display the feasible answer
+            displayBoard( chess );
+             //unvisit the cell
+            chess[r][c] = 0;
+            return;
+        }
+         //assign the jump number on the chess board
+        chess[r][c] = jump;
+         //recursive calls begins from here. It checks to see that the knight doesn't go out of the check board and also the cell should be unvisited
+        if( r - 2 >= 0 && c + 1 < chess[0].length && chess[ r-2 ][c+1] == 0 ){
+            knight( chess, r-2, c+1, jump+1 );
+        }
+        if( r - 1 >= 0 && c + 2 < chess[0].length && chess[ r-1 ][c+2] == 0 ){
+            knight( chess, r-1, c+2, jump+1 );
+        }
+        if( r + 1 < chess.length && c + 2 < chess[0].length && chess[ r+1 ][c+2] == 0 ){
+            knight( chess, r+1, c+2, jump+1 );
+        }
+        if( r + 2 < chess.length && c + 1 < chess[0].length && chess[ r+2 ][c+1] == 0 ){
+            knight( chess, r+2, c+1, jump+1 );
+        }
+        if( r + 2 < chess.length && c - 1 >= 0 && chess[ r+2 ][c-1] == 0 ){
+            knight( chess, r+2, c-1, jump+1 );
+        }
+        if( r + 1 < chess.length && c - 2 >= 0 && chess[ r+1 ][c-2] == 0 ){
+            knight( chess, r+1, c-2, jump+1 );
+        }
+        if( r - 1 >= 0 && c - 2  >= 0 && chess[ r-1 ][c-2] == 0 ){
+            knight( chess, r-1, c-2, jump+1 );
+        }
+        if( r - 2 >= 0 && c - 1 >= 0 && chess[ r-2 ][c-1] == 0 ){
+            knight( chess, r-2, c-1, jump+1 );
+        }
+        //unvisit the chess board. Backtracking here
+        chess[r][c] = 0;
+    }
+
+    public static void displayBoard(int[][] chess){
+        for(int i = 0; i < chess.length; i++){
+            for(int j = 0; j < chess[0].length; j++){
+                System.out.print(chess[i][j] + " ");
+            }
+            System.out.println();
+        }
+        System.out.println();
+    }
+}
+/* There are N^2 Cells and for each, we have a maximum of 8 possible moves to choose from, so the worst running time is O(8N^2). It prints all the feasible solutions (possibilities). When moving from (r, c) to the possible 8 options, given first precedence to (r - 2, c + 1) and 
```

Comment:
This looks much better however if you can please write time complexity separately would really be great!
Also, make sure that the comments should never be of such a big length that the reader has to scroll it sideways to read a whole of it.  Cut your comments small or whenever it gets big write the rest in the next line to improve readability.